server/base/src/test/java/org/apache/accumulo/server/util/time/SimpleTimerTest.java

```diff
@@ -98,11 +97,15 @@ public void testFailure() throws InterruptedException {
     assertTrue(r.wasRun);
   }
 
-  @Test
-  public void testSingleton() {
-    assertEquals(1, SimpleTimer.getInstanceThreadPoolSize());
-    SimpleTimer t2 = SimpleTimer.getInstance(2);
-    assertSame(t, t2);
+  @Test(timeout = 5000)
+  public void testSingleton() throws InterruptedException {
+    while (true) {
+      SimpleTimer t2 = SimpleTimer.getInstance(2);
+      if((SimpleTimer.getInstanceThreadPoolSize() == 1) && (t == t2)) {
+        break;
+      }
+      Thread.sleep(PAD);
+    }
```

Comment:
This is the right strategy (albeit with a few extra parenthesis than necessary on that `if` statement :wink: )

Reply:
Oops, I see I was off-base with this one. Will do my best to correct these issues with your suggestions in mind :+1: